src/main/java/seedu/duke/parser/Parser.java

```diff
@@ -1,11 +1,422 @@
+    public static final int EXIT_PROGRAM = 16;
+
+    // In-Module Commands
+    public static final int PRINT_HELP_MODULE = 21;
+    public static final int EXIT_MODULE = 22;
+    public static final int LIST_MODULE_INFO = 23;
+    public static final int LIST_LESSONS = 24;
+    public static final int OPEN_LINK = 25;
+    public static final int LIST_TASKS = 26;
+    public static final int MARK_DONE = 27;
+    public static final int UNMARK_DONE = 28;
+    public static final int LIST_TEACHING_STAFF = 29;
+    public static final int ADD_LESSON = 30;
+    public static final int DEL_LESSON = 31;
+    public static final int ADD_TASK = 32;
+    public static final int DEL_TASK = 33;
+    public static final int UNKNOWN_COMMAND = 99;
+    
+    // String Constants
+    public static final String DELIM = "\\s+;;\\s+";
+    public static final String WHITESPACE = " ";
+    
+    // Lesson parser constants
+    public static final int LESSON_TYPE_INDEX = 0;
+    public static final int LESSON_TIME_DAY_INDEX = 1;
+    public static final int LESSON_LINK_INDEX = 2;
+    public static final int LESSON_TEACHER_INDEX = 3;
+    public static final int LESSON_EMAIL_INDEX = 4;
+    
+    public static final int LESSON_MAX_DETAILS = 5;
+    
+    // Task parser constants
+    public static final int TASK_DESCRIPTION_INDEX = 0;
+    public static final int TASK_DEADLINE_INDEX = 1;
+    public static final int TASK_REMARKS_INDEX = 2;
+    
+    public static final int TASK_MAX_DETAILS = 3;
+
+    /**
+     * Calls the appropriate parser method depending on whether user is at dashboard or has selected
+     * a module.
+     *
+     * @param input full user input string
+     * @return command object based on user input
+     * @throws UnknownCommandException if valid command cannot be parsed from user input
+     */
+    public Command parse(String input) throws UnknownCommandException {
+        Command parsedCommand;
+
+        if (moduleIsSelected()) {
+            parsedCommand = parseInModule(input);
+        } else {
+            parsedCommand = parseAtDashboard(input);
+        }
+        return parsedCommand;
+    }
+
+    /**
+     * Parses dashboard commands from user input.
+     * User is yet to select a module.
+     *
+     * @param input full user input string
+     * @return dashboard command object based on user input
+     * @throws UnknownCommandException if valid command cannot be parsed from user input
+     */
+    private Command parseAtDashboard(String input) throws UnknownCommandException {
+        Command command;
+        int commandCode = parseDashboardCommandFromInput(input);
+        String moduleCode;
+
+        switch (commandCode) {
+        case ADD_MODULE:
+            moduleCode = getModuleCode(input);
+            command = new AddModuleCommand(moduleCode);
+            break;
+        case DEL_MODULE:
+            moduleCode = getModuleCode(input);
+            command = new DeleteModuleCommand(moduleCode);
```

Comment:
For the delete module command, are we deleting by index or by name?

Reply:
Just checked with Isa, no parameter needed as he will be getting user input for the index.